спринт13

SleeplessShu opened the pull request:
Переименовано:
- почти всё, в попытке придать удобную сортировку файлам

Добавлено:
- активити плеера

SleeplessShu committed:
Переименовано почти всё, в попытке придать удобную сортировку файлам

Добавлено:
- активити плеера

faritowich commented on app/src/main/java/com/practicum/playlistmaker_ver2/Activity_Base.kt:
```diff
@@ -9,7 +8,7 @@ import android.view.WindowManager
 import androidx.appcompat.app.AppCompatActivity
 
-open class BaseActivity : AppCompatActivity() {
+open class Activity_Base : AppCompatActivity() {
```
Через нижнее подчеркивание не принято называть классы. Но это конвенционально. ActivityBase. В других классах аналогично

SleeplessShu replied:
Привет! Очень радуюсь, когда ты ревьюешь мой код и даёшь советы. Спасибо большое, исправил.

faritowich commented on app/src/main/java/com/practicum/playlistmaker_ver2/Activity_Player.kt:
```diff
@@ -0,0 +1,98 @@
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_player)
+
+        val currentTrack: TrackData? = intent.getParcelableExtra("trackData")
```
"trackData" и вообще строки-ключи стоит выносить в отдельную константу (в данном случае в companion object). Это позволит не бояться забыть поменять в одном месте и забыть в другом. При необходимости просто меняешь в самой константе и всё так же работает. Плюс сразу понятно, где мы передаем и получаем эти данные.

SleeplessShu replied:
Спасибо за напоминание. поправил

faritowich commented on app/src/main/java/com/practicum/playlistmaker_ver2/Activity_Player.kt:
```diff
@@ -0,0 +1,98 @@
+            dp.toFloat(),
+            context.resources.displayMetrics
+        ).toInt()
+    }
```
Выглядит как довольно универсальная штука. Вынести бы её в отдельный файл

SleeplessShu replied:
сделал

faritowich commented on app/src/main/java/com/practicum/playlistmaker_ver2/Activity_Player.kt:
```diff
@@ -0,0 +1,98 @@
+        val minutes = trackTimeMillis / 1000 / 60
+        val seconds = (trackTimeMillis / 1000 % 60)
+        return String.format("%02d:%02d", minutes, seconds)
+    }
```
Да и это тоже универсально достаточно

SleeplessShu replied:
сделал

faritowich commented on app/src/main/java/com/practicum/playlistmaker_ver2/Activity_Player.kt:
```diff
@@ -0,0 +1,98 @@
+        val bBack = findViewById<ImageView>(R.id.bBack)
+        val bPlay = findViewById<ImageView>(R.id.bPlay)
+        val bLike = findViewById<ImageButton>(R.id.bLike)
+        val bAddToPlaylist = findViewById<ImageButton>(R.id.bAddToPlaylist)
```
Рекомендую почитать про библиотеку ViewBinding от гугла и использовать её - она используется повсеместно

SleeplessShu committed:
Изменены имена активити:
- убран символ "_"

Добавлено пролистывание в активити плеера:
- на маленьких устройствах есть возможность пролистать страницу до низа

Перенесены функции:
- общие функции вынесены в отдельный файл

Перенесены ключи:
- вынесены в компэнион обджект

findView заменена на View binding:
- во всех активити
